guides/source/7_1_release_notes.md

```diff
@@ -318,12 +318,31 @@ You can also set default values for these locals:
 <%= message %>
 ```
 
+Optional keyword arguments can be splatted:
+
+```erb
+<%# locals: (message: "Hello, world!", **attributes) -%>
+<%= tag.p(message, **attributes) %>
+```
+
 If you want to disable the use of locals entirely, you can do so like this:
 
 ```erb
 <%# locals: () %>
 ```
 
+All templating engines that support comments are supported:
+
+```ruby
+# locals: (json:, message:)
+json.message message
+```
```

Comment:
While this is true in the sense that Action View will parse out a `# locals: (...)` magic comment, engines like `jbuilder` merge additional values into the `locals:` Hash under keys like `:json`, `:collection`, or `:cached`).

This behavior is exhibited in the [test failures](https://github.com/rails/jbuilder/actions/runs/7642784095/job/20823237599?pr=557#step:4:62) included in https://github.com/rails/jbuilder/pull/557.

Instances of `ActionView::Template` can hint to the renderer a set of implicit keys that the templating engine is merging into the `:locals` Hash (for example, collection counters and the current iteration index) with its [`:implicit_locals` keyword argument](https://github.com/rails/rails/blob/c7551d08fc572655105da38394d8672b8259c22f/actionview/lib/action_view/template.rb#L248).

Unfortunately, templating engines like `jbuilder` are not integrating with Action View rendering at that level.

I see two paths forward for engines like `jbuilder`:

1. change their implementation to instead rely on template-scoped helper methods to provide access to baked-in references like the `json` object
  1a. alternatively, change their implementation to directly access an instance of `ActionView::Template` (maybe through the `@current_template` instance variable set in the `ActionView::Base` instance), then render with `ActionView::Template#render` instead of `ActionView::Base#render`)
2. change Action View's implementation to support passing in `:implicit_locals` from higher layers

I'm not sure of the labor involved in either approach, but it's probably worth considering, since Strict Locals has utility outside of ERB.

Reply:
pinging @byroot since you're the author of https://github.com/rails/rails/pull/49782, which resolved issues very similar to this one.

Reply:
@casperisfine do you have any insight here?

Reply:
@casperisfine for now, I've softened the language from `All templating engines that support comments are supported` to `Action View will process the `locals:` magic comment in any templating engine that supports `#`-prefixed comments`.

For now, that represents the truth, and we can investigate a public interface for templating engines separate from improving the documentation.